src/Worker/Grpc/GrpcDurableTaskWorker.Processor.cs

```diff
@@ -351,8 +351,75 @@ void RunBackgroundTask(P.WorkItem? workItem, Func<Task> handler)
                     string instanceId =
                         workItem?.OrchestratorRequest?.InstanceId ??
                         workItem?.ActivityRequest?.OrchestrationInstance?.InstanceId ??
+                        workItem?.EntityRequest?.InstanceId ??
+                        workItem?.EntityRequestV2?.InstanceId ??
                         string.Empty;
                     this.Logger.UnexpectedError(ex, instanceId);
+
+                    if (workItem?.OrchestratorRequest != null)
+                    {
+                        this.Logger.AbandoningOrchestratorWorkItem(instanceId, workItem?.CompletionToken ?? string.Empty);
+                        await this.client.AbandonTaskOrchestratorWorkItemAsync(
+                            new P.AbandonOrchestrationTaskRequest
+                            {
+                                CompletionToken = workItem?.CompletionToken,
+                            },
+                            cancellationToken: CancellationToken.None);
+                        this.Logger.AbandonedOrchestratorWorkItem(instanceId, workItem?.CompletionToken ?? string.Empty);
+                    }
+                    else if (workItem?.ActivityRequest != null)
+                    {
+                        this.Logger.AbandoningActivityWorkItem(
+                            instanceId,
+                            workItem.ActivityRequest.Name,
+                            workItem.ActivityRequest.TaskId,
+                            workItem?.CompletionToken ?? string.Empty);
+                        await this.client.AbandonTaskActivityWorkItemAsync(
```

Comment:
The abandon calls could themselves throw exeptions, which have to be caught because this is a background task. Also, we need to retry the 'Unavailable' grpc exceptions.

Reply:
updated, yes we have retry added to all grpc calls from this grpclient on unavaiable status code in the default gRPC retry policy but it is only applied when using the Azure Managed Durable Task Scheduler options src\Client\AzureManaged\DurableTaskSchedulerClientOptions.cs
src\Worker\AzureManaged\DurableTaskSchedulerWorkerOptions.cs